cross-modal-search/ClipImageEncoder/CLIPImageEncoder/__init__.py

```diff
@@ -0,0 +1,30 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import numpy as np
+from jina.executors.decorators import batching, as_ndarray
+from jina.executors.encoders.frameworks import BaseTorchEncoder
+from jina.executors.devices import TorchDevice
+import torch
+
+class CLIPImageEncoder(BaseTorchEncoder, TorchDevice):
+    
+    def __init__(self,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def post_init(self):
+        import clip
+
+        device = "cuda" if self.on_gpu else "cpu"
+        model, preprocess = clip.load('ViT-B/32', device)
```

Comment:
use `self.device` which actually detects

Reply:
add a `model_name` parameter so that it can choose which CLIP model can be used. ( we can check if the input is part of the `clip.available_models`

Reply:
CLIP now has `ViT-B/32` and `RN50` so someone may want to try different options